[BUG] test_hash_reduction_pivot_without_nans failed

Nightly Dataproc tests failed:
```
[2022-01-12T20:49:46.955Z] =================================== FAILURES ===================================
[2022-01-12T20:49:46.955Z] _ test_hash_reduction_pivot_without_nans[{'spark.rapids.sql.variableFloatAgg.enabled': 'true', 'spark.rapids.sql.hasNans': 'true', 'spark.rapids.sql.castStringToFloat.enabled': 'true', 'spark.rapids.sql.hashAgg.replaceMode': 'partial'}-[('a', Long), ('b', Integer), ('c', Long)]] _
[2022-01-12T20:49:46.955Z] 
[2022-01-12T20:49:46.955Z] data_gen = [('a', Long), ('b', Integer), ('c', Long)]
[2022-01-12T20:49:46.955Z] conf = {'spark.rapids.sql.castStringToFloat.enabled': 'true', 'spark.rapids.sql.hasNans': 'true', 'spark.rapids.sql.hashAgg.replaceMode': 'partial', 'spark.rapids.sql.variableFloatAgg.enabled': 'true'}
[2022-01-12T20:49:46.955Z] 
[2022-01-12T20:49:46.955Z]     @approximate_float
[2022-01-12T20:49:46.955Z]     @ignore_order(local=True)
[2022-01-12T20:49:46.955Z]     @incompat
[2022-01-12T20:49:46.955Z]     @pytest.mark.parametrize('data_gen', _init_list_no_nans, ids=idfn)
[2022-01-12T20:49:46.955Z]     @pytest.mark.parametrize('conf', get_params(_confs_with_nans, params_markers_for_confs_nans), ids=idfn)
[2022-01-12T20:49:46.955Z]     def test_hash_reduction_pivot_without_nans(data_gen, conf):
[2022-01-12T20:49:46.955Z] >       assert_gpu_and_cpu_are_equal_collect(
[2022-01-12T20:49:46.955Z]             lambda spark: gen_df(spark, data_gen, length=100)
[2022-01-12T20:49:46.955Z]                 .groupby()
[2022-01-12T20:49:46.955Z]                 .pivot('b')
[2022-01-12T20:49:46.955Z]                 .agg(f.sum('c')),
[2022-01-12T20:49:46.955Z]             conf=conf)
[2022-01-12T20:49:46.955Z] 
[2022-01-12T20:49:46.955Z] integration_tests/src/main/python/hash_aggregate_test.py:544: 
[2022-01-12T20:49:46.955Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[2022-01-12T20:49:46.955Z] integration_tests/src/main/python/asserts.py:505: in assert_gpu_and_cpu_are_equal_collect
[2022-01-12T20:49:46.955Z]     _assert_gpu_and_cpu_are_equal(func, 'COLLECT', conf=conf, is_cpu_first=is_cpu_first)
[2022-01-12T20:49:46.955Z] integration_tests/src/main/python/asserts.py:436: in _assert_gpu_and_cpu_are_equal
[2022-01-12T20:49:46.955Z]     assert_equal(from_cpu, from_gpu)
[2022-01-12T20:49:46.955Z] integration_tests/src/main/python/asserts.py:103: in assert_equal
[2022-01-12T20:49:46.955Z]     _assert_equal(cpu, gpu, float_check=get_float_check(), path=[])
[2022-01-12T20:49:46.955Z] integration_tests/src/main/python/asserts.py:42: in _assert_equal
[2022-01-12T20:49:46.955Z]     _assert_equal(cpu[index], gpu[index], float_check, path + [index])
[2022-01-12T20:49:46.955Z] integration_tests/src/main/python/asserts.py:35: in _assert_equal
[2022-01-12T20:49:46.955Z]     _assert_equal(cpu[field], gpu[field], float_check, path + [field])
[2022-01-12T20:49:46.955Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[2022-01-12T20:49:46.955Z] 
[2022-01-12T20:49:46.955Z] cpu = -3546155898819671139, gpu = 0
[2022-01-12T20:49:46.955Z] float_check = <function get_float_check.<locals>.<lambda> at 0x7f0bb2e388b0>
[2022-01-12T20:49:46.955Z] path = [0, '2117211837']
[2022-01-12T20:49:46.955Z] 
[2022-01-12T20:49:46.956Z]     def _assert_equal(cpu, gpu, float_check, path):
[2022-01-12T20:49:46.956Z]         t = type(cpu)
[2022-01-12T20:49:46.956Z]         if (t is Row):
[2022-01-12T20:49:46.956Z]             assert len(cpu) == len(gpu), "CPU and GPU row have different lengths at {} CPU: {} GPU: {}".format(path, len(cpu), len(gpu))
[2022-01-12T20:49:46.956Z]             if hasattr(cpu, "__fields__") and hasattr(gpu, "__fields__"):
[2022-01-12T20:49:46.956Z]                 assert cpu.__fields__ == gpu.__fields__, "CPU and GPU row have different fields at {} CPU: {} GPU: {}".format(path, cpu.__fields__, gpu.__fields__)
[2022-01-12T20:49:46.956Z]                 for field in cpu.__fields__:
[2022-01-12T20:49:46.956Z]                     _assert_equal(cpu[field], gpu[field], float_check, path + [field])
[2022-01-12T20:49:46.956Z]             else:
[2022-01-12T20:49:46.956Z]                 for index in range(len(cpu)):
[2022-01-12T20:49:46.956Z]                     _assert_equal(cpu[index], gpu[index], float_check, path + [index])
[2022-01-12T20:49:46.956Z]         elif (t is list):
[2022-01-12T20:49:46.956Z]             assert len(cpu) == len(gpu), "CPU and GPU list have different lengths at {} CPU: {} GPU: {}".format(path, len(cpu), len(gpu))
[2022-01-12T20:49:46.956Z]             for index in range(len(cpu)):
[2022-01-12T20:49:46.956Z]                 _assert_equal(cpu[index], gpu[index], float_check, path + [index])
[2022-01-12T20:49:46.956Z]         elif (t is tuple):
[2022-01-12T20:49:46.956Z]             assert len(cpu) == len(gpu), "CPU and GPU list have different lengths at {} CPU: {} GPU: {}".format(path, len(cpu), len(gpu))
[2022-01-12T20:49:46.956Z]             for index in range(len(cpu)):
[2022-01-12T20:49:46.956Z]                 _assert_equal(cpu[index], gpu[index], float_check, path + [index])
[2022-01-12T20:49:46.956Z]         elif (t is pytypes.GeneratorType):
[2022-01-12T20:49:46.956Z]             index = 0
[2022-01-12T20:49:46.956Z]             # generator has no zip :( so we have to do this the hard way
[2022-01-12T20:49:46.956Z]             done = False
[2022-01-12T20:49:46.956Z]             while not done:
[2022-01-12T20:49:46.956Z]                 sub_cpu = None
[2022-01-12T20:49:46.956Z]                 sub_gpu = None
[2022-01-12T20:49:46.956Z]                 try:
[2022-01-12T20:49:46.956Z]                     sub_cpu = next(cpu)
[2022-01-12T20:49:46.956Z]                 except StopIteration:
[2022-01-12T20:49:46.956Z]                     done = True
[2022-01-12T20:49:46.956Z]     
[2022-01-12T20:49:46.956Z]                 try:
[2022-01-12T20:49:46.956Z]                     sub_gpu = next(gpu)
[2022-01-12T20:49:46.956Z]                 except StopIteration:
[2022-01-12T20:49:46.956Z]                     done = True
[2022-01-12T20:49:46.956Z]     
[2022-01-12T20:49:46.956Z]                 if done:
[2022-01-12T20:49:46.956Z]                     assert sub_cpu == sub_gpu and sub_cpu == None, "CPU and GPU generators have different lengths at {}".format(path)
[2022-01-12T20:49:46.956Z]                 else:
[2022-01-12T20:49:46.956Z]                     _assert_equal(sub_cpu, sub_gpu, float_check, path + [index])
[2022-01-12T20:49:46.956Z]     
[2022-01-12T20:49:46.956Z]                 index = index + 1
[2022-01-12T20:49:46.956Z]         elif (t is dict):
[2022-01-12T20:49:46.956Z]             # The order of key/values is not guaranteed in python dicts, nor are they guaranteed by Spark
[2022-01-12T20:49:46.956Z]             # so sort the items to do our best with ignoring the order of dicts
[2022-01-12T20:49:46.956Z]             cpu_items = list(cpu.items()).sort(key=_RowCmp)
[2022-01-12T20:49:46.956Z]             gpu_items = list(gpu.items()).sort(key=_RowCmp)
[2022-01-12T20:49:46.956Z]             _assert_equal(cpu_items, gpu_items, float_check, path + ["map"])
[2022-01-12T20:49:46.956Z]         elif (t is int):
[2022-01-12T20:49:46.956Z] >           assert cpu == gpu, "GPU and CPU int values are different at {}".format(path)
[2022-01-12T20:49:46.956Z] E           AssertionError: GPU and CPU int values are different at [0, '2117211837']
[2022-01-12T20:49:46.956Z] 
[2022-01-12T20:49:46.956Z] integration_tests/src/main/python/asserts.py:77: AssertionError
```

--- a/integration_tests/src/main/python/hash_aggregate_test.py
+++ b/integration_tests/src/main/python/hash_aggregate_test.py
@@ -533,8 +533,6 @@ def test_hash_pivot_reduction_nan_fallback(data_gen):
         "PivotFirst",
         conf=_nans_float_conf)
 
-@pytest.mark.xfail(reason="Disabling below test temporarily until we have a fix for this issue "
-                          "https://github.com/NVIDIA/spark-rapids/issues/4514")
 @approximate_float
 @ignore_order(local=True)
 @incompat

